src/openrct2-ui/audio/SDLAudioSource.cpp

```diff
@@ -116,46 +116,42 @@ std::unique_ptr<SDLAudioSource> OpenRCT2::Audio::CreateAudioSource(SDL_RWops* rw
 std::unique_ptr<SDLAudioSource> OpenRCT2::Audio::CreateAudioSource(SDL_RWops* rw, uint32_t cssIndex)
 {
     auto numSounds = SDL_ReadLE32(rw);
-    if (cssIndex < numSounds)
+    if (cssIndex >= numSounds)
     {
-        SDL_RWseek(rw, cssIndex * 4, RW_SEEK_CUR);
-
-        auto pcmOffset = SDL_ReadLE32(rw);
-        SDL_RWseek(rw, pcmOffset, RW_SEEK_SET);
-
-        auto pcmLength = SDL_ReadLE32(rw);
-
-        AudioFormat format;
-        [[maybe_unused]] auto encoding = SDL_ReadLE16(rw);
-        format.channels = SDL_ReadLE16(rw);
-        format.freq = SDL_ReadLE32(rw);
-        [[maybe_unused]] auto byterate = SDL_ReadLE32(rw);
-        [[maybe_unused]] auto blockalign = SDL_ReadLE16(rw);
-        [[maybe_unused]] auto bitspersample = SDL_ReadLE16(rw);
-        switch (bitspersample)
-        {
-            case 8:
-                format.format = AUDIO_U8;
-                break;
-            case 16:
-                format.format = AUDIO_S16LSB;
-                break;
-            default:
-                SDL_RWclose(rw);
-                throw std::runtime_error("Unsupported bits per sample");
-        }
-        [[maybe_unused]] auto extrasize = SDL_ReadLE16(rw);
-
-        std::vector<uint8_t> pcmData;
-        pcmData.resize(pcmLength);
-        SDL_RWread(rw, pcmData.data(), pcmLength, 1);
-
-        SDL_RWclose(rw);
-        return CreateMemoryAudioSource(format, format, std::move(pcmData));
+        // Not enough sounds, caller is responsible for freeing rw
+        return nullptr;
     }
-    else
+
+    SDL_RWseek(rw, cssIndex * 4, RW_SEEK_CUR);
+
+    auto pcmOffset = SDL_ReadLE32(rw);
+    SDL_RWseek(rw, pcmOffset, RW_SEEK_SET);
+
+    auto pcmLength = SDL_ReadLE32(rw);
+
+    AudioFormat format;
+    [[maybe_unused]] auto encoding = SDL_ReadLE16(rw);
+    format.channels = SDL_ReadLE16(rw);
+    format.freq = SDL_ReadLE32(rw);
+    [[maybe_unused]] auto byterate = SDL_ReadLE32(rw);
+    [[maybe_unused]] auto blockalign = SDL_ReadLE16(rw);
+    [[maybe_unused]] auto bitspersample = SDL_ReadLE16(rw);
```

Comment:
What is point of those variables if they aren't actually used? For release builds those will vanish in case the intention is to have them in a stack frame.

Reply:
They come from here: https://github.com/janisozaur/OpenRCT2/commit/95661f4df2f487cca26dbc0a3013a8024a426052

I guess the idea was to support them at some point?

Reply:
Done